infrastructure/builder/src/build/service/web-ui-nginx-site.conf

```diff
@@ -34,5 +34,6 @@ server {
 
   location @rewrites {
     rewrite ^(.+)$ /index.html last;
+    rewrite ^\/docs/$ /docs.html last;
```

Comment:
> the docs page just keeps reloading in a loop https://owlish.imbstack.com/docs/

Here's what I think is happening:

nginx is not serving `docs.html` when the user types `https://owlish.imbstack.com/docs` in the URL bar. It's serving `index.html`. `index.html` has a [react-router route](https://github.com/taskcluster/taskcluster-web/blob/ade83943bdf93957f8e55c30c9dfd62e0a4cc536/src/App/routes.jsx#L6-L10) which tries to switch entry points whenever the user navigates to a route from `index.html` that starts with `/docs` (for example clicking a link in tc-web that should navigate the user to the docs site). This calls the [SwitchEntryPoint](https://github.com/taskcluster/taskcluster-web/blob/ade83943bdf93957f8e55c30c9dfd62e0a4cc536/src/views/SwitchEntryPoint/index.jsx) component which reloads the page but since nginx fails to route to `docs.html` after reloading, it continuously refreshes the page.

Reply:
In production, I am using the following rewrite rules on the netlify site: https://github.com/taskcluster/taskcluster-web/blob/ade83943bdf93957f8e55c30c9dfd62e0a4cc536/netlify.toml#L12-L21. Hope that  helps to get the rewrites work with nginx.

Reply:
Netlify has such a wonderful user interface there, comparing to nginx's regexes :( Thank you Hassan! I'll correct my regexes